### DISCUSSION TITLE
Expanding OnDeleteAction enum by adding SetNull

### DISCUSSION THREAD

**Discussion by radiy** [nhibernate/nhibernate-core#2629] [Ideas]
Hi, what about adding SetNull to https://github.com/nhibernate/nhibernate-core/blob/ac39173567b31bcfad475ca32687c9faf0d37f87/src/NHibernate/Mapping/ByCode/IKeyMapper.cs#L7.
Any particular reason for not doing it?

**Comment by fredericDelaporte** [comment]
It just needs to be actually implemented, starting from hbm through everywhere `on-delete` action is implemented.

**Reply by radiy** [reply]
Totally agree. It was a bit of surprise to find it unimplemented.
I was thinking that maybe there some special reasons for this.
What do you think about implementation strategy?
It mapped to bool value https://github.com/nhibernate/nhibernate-core/blob/58e3cb220782ebc308a568e7c6cb4e6605f067d4/src/NHibernate/Mapping/ForeignKey.cs#L69 and changing it to enum will be breaking change, but creating another bool field feels unnatural.
I wonder what would be better, creating new boolean field  `bool CascadeSetNullEnabled` or creating new enum field `DeleteAction CascadDeleteAction` and obsoleting CascadeDeleteEnabled.

**Comment by fredericDelaporte** [comment]
> creating new enum field `DeleteAction CascadDeleteAction` and obsoleting `CascadeDeleteEnabled`

I would go for this, with `CascadeDeleteEnabled` just yielding its value according to `CascadDeleteAction` value and setting `CascadDeleteAction` on set.
